### PR TITLE
SDL2: Fetch Through CMake

### PR DESCRIPTION
**Pull request by HTRamsey**
Fetches SDL2 through cmake and just builds the necessary components for joystick. This will make all platforms run the same version. When qmake goes, the libs/sdl2/msvc and Frameworks/SDL2.framework stuff can be deleted.

This means ios and android could use sdl2 as well

### CODE REVIEW

File: cmake/BuildSdl2.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# find_package(SDL2 QUIET)
+if(SDL2_FOUND)
```
Comment (DonLakeFlyer) [inline cmake/BuildSdl2.cmake:2]: This seems odd. the find_package is commented out. Won't this always the false?

File: cmake/BuildSdl2.cmake [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    message(STATUS "Using SDL2 via find_package")
+endif()
+
+# 2. Try using a vendored SDL library
+if(NOT SDL2_FOUND AND EXISTS "${CMAKE_CURRENT_LIST_DIR}/SDL/CMakeLists.txt")
```
Comment (DonLakeFlyer) [inline cmake/BuildSdl2.cmake:7]: Why is this needed?

Comment (HTRamsey) [reply cmake/BuildSdl2.cmake:7]: It's not, this was while I was waiting to figure out how we were going to handle submodules. Same with the other issue.
